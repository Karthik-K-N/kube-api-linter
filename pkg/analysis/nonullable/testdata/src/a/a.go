package a

// +nullable
type NullableMarkerType string // want `type NullableMarkerType has forbidden marker "nullable"`

// +allowed
type AllowedMarkerType string

type Test struct {
	// +nullable
	NullableMarkerField string `json:"nullableMarkerField"`// want `field NullableMarkerField has forbidden marker "nullable"`

	NullableMarkerFieldTypeAlias NullableMarkerType `json:"nullableMarkerFieldTypeAlias"` // want `field NullableMarkerFieldTypeAlias has forbidden marker "nullable"`

	// +allowed
	AllowedMarkerField string `json:"allowedMarkerField"`

	AllowedMarkerFieldTypeAlias AllowedMarkerType `json:"AllowedMarkerFieldTypeAlias"`
}

