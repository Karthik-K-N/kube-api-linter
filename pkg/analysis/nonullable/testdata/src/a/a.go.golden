package a

type NullableMarkerType string // want `type NullableMarkerType has forbidden marker "nullable"`

// +allowed
type AllowedMarkerType string

type Test struct {
	NullableMarkerField string `json:"nullableMarkerField"`// want `field NullableMarkerField has forbidden marker "nullable"`

	NullableMarkerFieldTypeAlias NullableMarkerType `json:"nullableMarkerFieldTypeAlias"` // want `field NullableMarkerFieldTypeAlias has forbidden marker "nullable"`

	// +allowed
	AllowedMarkerField string `json:"allowedMarkerField"`

	AllowedMarkerFieldTypeAlias AllowedMarkerType `json:"AllowedMarkerFieldTypeAlias"`
}

